src/components/general/ButtonLayout.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { View, Dimensions } from "react-native";
+
+const dimension = Dimensions.get("window");
+
+class ButtonLayout extends Component {
+	constructor(props) {
+		super(props);
+	}
+
+	static propTypes = {
+		icon: PropTypes.shape({})
+	}
+
+	render() {
+		const {
+			buttonContainer,
+			doubleLayer
+		} = styles;
+
+		const {
+			icon,
+			children,
+			containerStyle
+		} = this.props;
+
+		let i = 0;
+		let realButtons = [];
+		let length;
+		let layers = [];
+		let iconVar;
+
+		if (!children) {
+			return null;
+		}
+		else if (!children.length) {
+			if (children) {
```

Comment:
Consider joining these in a single condition, if possible.

Reply:
Currently could not find a way to avoid having another if statement